pandas/core/tools/datetimes.py

```diff
@@ -200,19 +200,27 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
         if format is not None:
             raise ValueError("cannot specify both format and unit")
         arg = getattr(arg, 'values', arg)
-        result = tslib.array_with_unit_to_datetime(arg, unit,
-                                                   errors=errors)
+        result, tz_parsed = tslib.array_with_unit_to_datetime(arg, unit,
+                                                              errors=errors)
         if box:
             if errors == 'ignore':
                 from pandas import Index
                 result = Index(result, name=name)
-                # GH 23758: We may still need to localize the result with tz
-                try:
-                    return result.tz_localize(tz)
-                except AttributeError:
-                    return result
-
-            return DatetimeIndex(result, tz=tz, name=name)
+            else:
+                result = DatetimeIndex(result, name=name)
+            # GH 23758: We may still need to localize the result with tz
+            # GH 25546: Apply tz_parsed first (from arg), then tz (from caller)
+            # result will be naive but in UTC
+            try:
+                result = result.tz_localize('UTC').tz_convert(tz_parsed)
+            except AttributeError:
+                # Regular Index from 'ignore' path
```

Comment:
how could this ever be hit as you are constructing a DTI here?

further should not all of the localization *only* be on this path (e.g. you must have a DTI); the way its written you *could* be an Index? (which is not possible)

Reply:
Above there's 

```
 if errors == 'ignore':
    from pandas import Index
    result = Index(result, name=name)
```

Where this _could_ be an `Index` if there were actually errors or coerced to `DatetimeIndex` if there were no errors.

Reply:
right so what hits that? meaning if it is not coerced, then we can *never* attempt the localization no matter what

Reply:
Yeah if it's not coerced to `DatetimeIndex` we're just returning the result (`Index`). `array_with_unit_to_datetime` above could return an object array that wouldn't get coerced to `DatetimeIndex`